Glue unified catalog should not required hive URI

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
I am trying out the unreleased unified catalog feature, using a custom build of Starrocks.

```
CREATE EXTERNAL CATALOG unified_catalog
PROPERTIES
(
    "type" = "unified",
    "unified.metastore.type" = "glue",
    "aws.glue.use_aws_sdk_default_behavior" = "true",
    "aws.glue.region" = "us-west-2",
    "aws.s3.use_aws_sdk_default_behavior" = "true",
    "aws.s3.region" = "us-west-2",
    "enable_metastore_cache" = "true",
    "enable_remote_file_cache" = "false"
);
```

### Expected behavior (Required)
Able to create a unified catalog with AWS Glue.

### Real behavior (Required)
Creation errors out.

SQL client reports `Query 1 ERROR: Unexpected exception: connector create failed`.

FE logs show that `hive.metastore.uris` is required, but this should not be required for an AWS Glue catalog.

```
2023-10-06 22:58:46,605 ERROR (starrocks-mysql-nio-pool-2|229) [ConnectorFactory.createConnector():56] can't create connector for type: unified
java.lang.reflect.InvocationTargetException: null
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
	at com.starrocks.connector.ConnectorFactory.createConnector(ConnectorFactory.java:43) ~[starrocks-fe.jar:?]
	at com.starrocks.connector.ConnectorMgr.createConnector(ConnectorMgr.java:36) ~[starrocks-fe.jar:?]
	at com.starrocks.server.CatalogMgr.createCatalog(CatalogMgr.java:116) ~[starrocks-fe.jar:?]
	at com.starrocks.server.CatalogMgr.createCatalog(CatalogMgr.java:97) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.DDLStmtExecutor$StmtExecutorVisitor.lambda$visitCreateCatalogStatement$70(DDLStmtExecutor.java:844) ~[starrocks-fe.jar:?]
	at com.starrocks.common.ErrorReport.wrapWithRuntimeException(ErrorReport.java:112) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.DDLStmtExecutor$StmtExecutorVisitor.visitCreateCatalogStatement(DDLStmtExecutor.java:843) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.DDLStmtExecutor$StmtExecutorVisitor.visitCreateCatalogStatement(DDLStmtExecutor.java:156) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.ast.CreateCatalogStmt.accept(CreateCatalogStmt.java:63) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.DDLStmtExecutor.execute(DDLStmtExecutor.java:142) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.handleDdlStmt(StmtExecutor.java:1492) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:658) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:377) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:569) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:848) ~[starrocks-fe.jar:?]
	at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.lang.NullPointerException: hive.metastore.uris must be set in properties when creating hive catalog
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:1010) ~[spark-dpp-1.0.0.jar:?]
	at com.starrocks.connector.delta.DeltaLakeInternalMgr.(DeltaLakeInternalMgr.java:50) ~[starrocks-fe.jar:?]
	at com.starrocks.connector.delta.DeltaLakeConnector.(DeltaLakeConnector.java:47) ~[starrocks-fe.jar:?]
	at com.starrocks.connector.unified.UnifiedConnector.(UnifiedConnector.java:60) ~[starrocks-fe.jar:?]
	... 23 more
```

### StarRocks version (Required)
Built from `main` on October 6th.

`main-d26ad8b`

--- a/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeConnector.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeConnector.java
@@ -15,8 +15,6 @@
 
 package com.starrocks.connector.delta;
 
-import com.google.common.base.Preconditions;
-import com.starrocks.common.util.Util;
 import com.starrocks.connector.Connector;
 import com.starrocks.connector.ConnectorContext;
 import com.starrocks.connector.ConnectorMetadata;
@@ -47,14 +45,6 @@ public class DeltaLakeConnector implements Connector {
         this.internalMgr = new DeltaLakeInternalMgr(catalogName, properties, hdfsEnvironment);
         this.metadataFactory = createMetadataFactory();
         // TODO extract to ConnectorConfigFactory
-        validate();
-        onCreate();
-    }
-
-    public void validate() {
-        String hiveMetastoreUris = Preconditions.checkNotNull(properties.get(HIVE_METASTORE_URIS),
-                "%s must be set in properties when creating hive catalog", HIVE_METASTORE_URIS);
-        Util.validateMetastoreUris(hiveMetastoreUris);
     }
 
     @Override
@@ -68,13 +58,10 @@ public class DeltaLakeConnector implements Connector {
                 catalogName,
                 metastore,
                 internalMgr.getHiveMetastoreConf(),
-                properties.get(HIVE_METASTORE_URIS),
+                properties,
                 internalMgr.getHdfsEnvironment(),
                 internalMgr.getMetastoreType()
         );
-    }
-
-    public void onCreate() {
     }
 
     public CloudConfiguration getCloudConfiguration() {

--- a/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeInternalMgr.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeInternalMgr.java
@@ -15,6 +15,7 @@
 package com.starrocks.connector.delta;
 
 import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import com.starrocks.common.util.Util;
 import com.starrocks.connector.HdfsEnvironment;
@@ -22,24 +23,28 @@ import com.starrocks.connector.MetastoreType;
 import com.starrocks.connector.ReentrantExecutor;
 import com.starrocks.connector.hive.CachingHiveMetastore;
 import com.starrocks.connector.hive.CachingHiveMetastoreConf;
+import com.starrocks.connector.hive.HiveConnector;
 import com.starrocks.connector.hive.HiveMetaClient;
 import com.starrocks.connector.hive.HiveMetastore;
 import com.starrocks.connector.hive.IHiveMetastore;
+import com.starrocks.sql.analyzer.SemanticException;
 
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
-import static com.starrocks.connector.delta.DeltaLakeConnector.HIVE_METASTORE_URIS;
+import static com.starrocks.connector.hive.HiveConnector.HIVE_METASTORE_TYPE;
 
 public class DeltaLakeInternalMgr {
+    public static final List<String> SUPPORTED_METASTORE_TYPE = ImmutableList.of("hive", "glue");
     private final String catalogName;
     private final Map<String, String> properties;
     private final HdfsEnvironment hdfsEnvironment;
     private final boolean enableMetastoreCache;
     private final CachingHiveMetastoreConf hmsConf;
     private ExecutorService refreshHiveMetastoreExecutor;
-    private MetastoreType metastoreType = MetastoreType.HMS;
+    private final MetastoreType metastoreType;
 
     public DeltaLakeInternalMgr(String catalogName, Map<String, String> properties, HdfsEnvironment hdfsEnvironment) {
         this.catalogName = catalogName;
@@ -47,9 +52,18 @@ public class DeltaLakeInternalMgr {
         this.enableMetastoreCache = Boolean.parseBoolean(properties.getOrDefault("enable_metastore_cache", "false"));
         this.hmsConf = new CachingHiveMetastoreConf(properties, "delta lake");
         this.hdfsEnvironment = hdfsEnvironment;
-        String hiveMetastoreUris = Preconditions.checkNotNull(properties.get(HIVE_METASTORE_URIS),
-                "%s must be set in properties when creating hive catalog", HIVE_METASTORE_URIS);
-        Util.validateMetastoreUris(hiveMetastoreUris);
+
+        String hiveMetastoreType = properties.getOrDefault(HIVE_METASTORE_TYPE, "hive").toLowerCase();
+        if (!SUPPORTED_METASTORE_TYPE.contains(hiveMetastoreType)) {
+            throw new SemanticException("hive metastore type [%s] is not supported", hiveMetastoreType);
+        }
+
+        if (hiveMetastoreType.equals("hive")) {
+            String hiveMetastoreUris = Preconditions.checkNotNull(properties.get(HiveConnector.HIVE_METASTORE_URIS),
+                    "%s must be set in properties when creating hive catalog", HiveConnector.HIVE_METASTORE_URIS);
+            Util.validateMetastoreUris(hiveMetastoreUris);
+        }
+        this.metastoreType = MetastoreType.get(hiveMetastoreType);
     }
 
     public IHiveMetastore createHiveMetastore() {

--- a/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeInternalMgr.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeInternalMgr.java
@@ -23,7 +23,6 @@ import com.starrocks.connector.MetastoreType;
 import com.starrocks.connector.ReentrantExecutor;
 import com.starrocks.connector.hive.CachingHiveMetastore;
 import com.starrocks.connector.hive.CachingHiveMetastoreConf;
-import com.starrocks.connector.hive.HiveConnector;
 import com.starrocks.connector.hive.HiveMetaClient;
 import com.starrocks.connector.hive.HiveMetastore;
 import com.starrocks.connector.hive.IHiveMetastore;
@@ -35,6 +34,7 @@ import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
 import static com.starrocks.connector.hive.HiveConnector.HIVE_METASTORE_TYPE;
+import static com.starrocks.connector.hive.HiveConnector.HIVE_METASTORE_URIS;
 
 public class DeltaLakeInternalMgr {
     public static final List<String> SUPPORTED_METASTORE_TYPE = ImmutableList.of("hive", "glue");
@@ -59,8 +59,8 @@ public class DeltaLakeInternalMgr {
         }
 
         if (hiveMetastoreType.equals("hive")) {
-            String hiveMetastoreUris = Preconditions.checkNotNull(properties.get(HiveConnector.HIVE_METASTORE_URIS),
-                    "%s must be set in properties when creating hive catalog", HiveConnector.HIVE_METASTORE_URIS);
+            String hiveMetastoreUris = Preconditions.checkNotNull(properties.get(HIVE_METASTORE_URIS),
+                    "%s must be set in properties when creating catalog of hive-metastore", HIVE_METASTORE_URIS);
             Util.validateMetastoreUris(hiveMetastoreUris);
         }
         this.metastoreType = MetastoreType.get(hiveMetastoreType);

--- a/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadataFactory.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadataFactory.java
@@ -22,6 +22,9 @@ import com.starrocks.connector.hive.HiveMetastoreOperations;
 import com.starrocks.connector.hive.IHiveMetastore;
 import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
 
+import java.util.Map;
+
+import static com.starrocks.connector.delta.DeltaLakeConnector.HIVE_METASTORE_URIS;
 import static com.starrocks.connector.hive.CachingHiveMetastore.createQueryLevelInstance;
 
 public class DeltaLakeMetadataFactory {
@@ -32,12 +35,16 @@ public class DeltaLakeMetadataFactory {
     private final MetastoreType metastoreType;
 
     public DeltaLakeMetadataFactory(String catalogName, IHiveMetastore metastore, CachingHiveMetastoreConf hmsConf,
-                                    String uri, HdfsEnvironment hdfsEnvironment, MetastoreType metastoreType) {
+                                    Map<String, String> properties, HdfsEnvironment hdfsEnvironment,
+                                    MetastoreType metastoreType) {
         this.catalogName = catalogName;
         this.metastore = metastore;
         this.perQueryMetastoreMaxNum = hmsConf.getPerQueryCacheMaxNum();
         this.hdfsEnvironment = hdfsEnvironment;
-        this.hdfsEnvironment.getConfiguration().set(MetastoreConf.ConfVars.THRIFT_URIS.getHiveName(), uri);
+        if (properties.containsKey(HIVE_METASTORE_URIS)) {
+            this.hdfsEnvironment.getConfiguration().set(MetastoreConf.ConfVars.THRIFT_URIS.getHiveName(),
+                    properties.get(HIVE_METASTORE_URIS));
+        }
         this.metastoreType = metastoreType;
     }
 

